includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -803,11 +785,6 @@ private function is_product_supported( $product ) {
 			return false;
 		}
 
-		// Not supported for subscription products when user is not authenticated and account creation is not possible.
-		if ( class_exists( 'WC_Subscriptions_Product' ) && WC_Subscriptions_Product::is_subscription( $product ) && ! is_user_logged_in() && ! $this->is_account_creation_possible() ) {
-			return false;
-		}
-
```

Comment:
Apart from that. This PR tries to enable the PR button on the product page so I need to remove the conditional. Maybe I can adjust the conditional but I didn't got any issues by removing it.

Reply:
> In fact the PR button is not displayed to the user when authentication is required.

That's because [we're preventing it](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/cb456e28d1af5eaa272603f60a5ea415aa094d4e/includes/payment-methods/class-wc-stripe-payment-request.php#L735-L737) when both guest checkout + creating an account upon checkout are not possible.

Settings to reproduce:

<img width="806" alt="Screen Shot 2021-06-21 at 20 03 26" src="https://user-images.githubusercontent.com/5509901/122838300-2b817d80-d2cc-11eb-943a-6d0454b850ac.png">

Error:

<img width="827" alt="Screen Shot 2021-06-21 at 20 04 21" src="https://user-images.githubusercontent.com/5509901/122838309-320ff500-d2cc-11eb-962a-2d4e75720c63.png">

> This PR tries to enable the PR button on the product page so I need to remove the conditional.

If you read the last part of the check, it's not about the subscription product, it's about disallowing the PRB for unauthenticated users when account creation is not possible, so we prevent the aforementioned error:

```php
// Not supported for subscription products when user is not authenticated and account creation is not possible.
if ( class_exists( 'WC_Subscriptions_Product' ) && WC_Subscriptions_Product::is_subscription( $product ) && ! is_user_logged_in() && ! $this->is_account_creation_possible() ) {
	return false;
}
```

As you know, we're refactoring how we deal with this error in #1609.

With that said, by reverting this check, maybe we should rename the PR to something related to product variations so it's more appropriate and doesn't get confused when searching. Feel free to ignore.

Reply:
@ricardo  I've tried multiple ways to reproduce the error including setting the options like the image you shared but I get no error. Am I missing something? How do you get the error? When clicking the PR button? What types of subscription do you have?

Reply:
Are you logged in while testing? You need to be logged out. The error happens when you try to pay using PRB. I tested with a simple subscription, but I think it can be any supported subscription.

Reply:
@ricardo yes. I'm logged out. I've even tried with an incognito window.

Reply:
I've updated this branch with the latest changes from `trunk` and this discussion seems no longer relevant. The PR button now shows an alert when the user is trying to purchase a subscription. 

<img width="693" alt="Screen Shot 2021-06-21 at 11 54 15 PM" src="https://user-images.githubusercontent.com/1025173/122865413-fd139a80-d2eb-11eb-9083-8c50afc35038.png">